bokeh/document.py

```diff
@@ -298,6 +300,12 @@ def __init__(self, **kwargs):
         self._callbacks = {}
         self._session_callbacks = {}
         self._session_context = None
+        self._modules = []
+
+    def __del__(self):
+        for module in self._modules:
+            if module.__name__ in sys.modules:
+                del sys.modules[module.__name__]
```

Comment:
And I guess an explanation of purpose could help too. For instance, if there are multiple docs with the same module in `doc._modules`, shouldn't it stay in `sys.modules` until all the docs are gone? (I'm probably missing something here)